### PR TITLE
feat: scrolling to option in select on keypress

### PR DESCRIPTION
**Pull request by robertbrada**
Fix the remaining part of https://github.com/trezor/trezor-suite/issues/2480

I used this feature in Language, Currency and NetworkSelect

### CODE REVIEW

File: packages/components/src/components/form/Select/index.tsx
Comment (slowbackspace) [inline packages/components/src/components/form/Select/index.tsx]: any boooo :D, let's figure out proper type

File: packages/components/src/components/form/Select/index.tsx
Comment (slowbackspace) [inline packages/components/src/components/form/Select/index.tsx:309]: big thumbs up for these comments!

File: packages/components/src/components/form/Select/index.tsx
Comment (slowbackspace) [inline packages/components/src/components/form/Select/index.tsx]: todooo

File: packages/components/src/components/form/Select/index.tsx
Comment (slowbackspace) [inline packages/components/src/components/form/Select/index.tsx]: hmm this could be probably handled by some typescript magic (props could be union of 2 types, one type would include something like `useKeyPressScroll: true; isSearchable: never`, the other type would be `isSearchable: true; useKeyPressScroll: never `). I am not completely sure on the syntax... anyway this is just nitpicking, but if you would like to try it then go ahead :)

https://github.com/trezor/trezor-suite/blob/50a59439d820d9e73a445a2f9a6d17772d498103/packages%2Fsuite%2Fsrc%2Fcomponents%2Fsuite%2FFiatValue%2FContainer.ts

**Review by slowbackspace — CHANGES_REQUESTED** [review]

**Comment by slowbackspace** [comment]
I think it is implemented fully to the specification that was provided, but I can't merge this in a good faith as I find the new behaviour really annoying and against all good UX practises.
One example:
Imagine you want to add doge account so you click on add account, modal appears, click on select and you start typing (as any person that used computers in the past 20 or so years would do): D O G => you end up with Dash preselected, you will say to yourself hmm maybe I did something wrong. So you start typing again D O G E => Ethereum will get selected. This is not how literally every select in every OS and environment works. I find it unacceptable and will fight for it if necessary  😁

update: we've come to agreement that we'll implement a behaviour of native selects, solving issue mentioned above 🙇

File: packages/components/src/components/form/Select/index.tsx
Comment (keraf) [inline packages/components/src/components/form/Select/index.tsx]: Proper type here would be welcome. Also you can omit `React.` and simply use `useRef` as it is imported above.

**Review by keraf — CHANGES_REQUESTED** [review]

File: packages/components/src/components/form/Select/index.tsx
Comment (slowbackspace) [inline packages/components/src/components/form/Select/index.tsx]: I would keep it under 500ms to better simulate native behaviour

File: packages/components/src/components/form/Select/index.tsx
Comment (slowbackspace) [inline packages/components/src/components/form/Select/index.tsx:292]: hmm shouldn't we reset search string only after the period of inactivity regardless of whether there is a match?

Comment (robertbrada) [reply packages/components/src/components/form/Select/index.tsx:292]: Was not sure about this, because then the user can type fast "_dsvnsivndsopjvioosvv..._" forever and the searchedTerm gets never cleared.  But yeah, the reality is that the user will never type "_dsvnsivndsopjvioosvv...._"

I thought it might be a good idea to clear the term once it's clear there will be no match no matter what the user types. But I am not sure what's the best practice here.

**Review by slowbackspace — APPROVED** [review]
lgtm, let's see if we'll hit some issue after using it in develop for a while 😁

**Pull request merged**
